File: 06_1-Balanceo_por_parentesis/python/main.py
from stackNode import StackNode

def readFileContent(fileName):
    file = open(fileName, "r")
    return file.read()

def checkParenthesis(stringToValidate):
    validatorStack = StackNode()
    for currentChar in stringToValidate:
        if currentChar == "(" or currentChar == "[" or currentChar == "{":
            validatorStack.push(currentChar)
        elif currentChar == ")" or currentChar == "]" or currentChar == "}":
            if validatorStack.isEmpty():
                return False
            else:
                charInStack = validatorStack.pop()
                if currentChar == ')' and charInStack != '(':
                    return False;
                elif currentChar == ']' and charInStack != '[':
                    return False;
                elif currentChar == '}' and charInStack != '{':
                    return False;
    return validatorStack.isEmpty()


file1 = "texto1.txt";
contentFile1 = readFileContent(file1);
balancerFile1 = checkParenthesis(contentFile1);
print("Archivo " + file1 + " - Balanceo: " + str(balancerFile1));

file2 = "texto2.txt";
contentFile2 = readFileContent(file2);
balancerFile2 = checkParenthesis(contentFile2);
print("Archivo " + file2 + " - Balanceo: " + str(balancerFile2));

File: 06_1-Balanceo_por_parentesis/cpp/main.cpp
#include "stackNode.h"
#include <iostream>
#include <fstream>
#include <string>
#include <sstream>
#include <vector>

using namespace std;

string readFileContent(string fileName) {
    string line, fileContent = "";
    fstream file(fileName, ios::in);
    if(file.is_open()) {
        while(getline(file, line)) {
            fileContent += line + "\n";
        }
        file.close();
    } else {
        cout<<"No se pudo abrir el archivo\n";
    }
    return fileContent;
}

bool checkParenthesis(string stringToValidate) {
    stackNode<char> *validatorStack = new stackNode<char>();
    for(int i = 0; i < stringToValidate.length(); i++) {
        char currentChar = stringToValidate[i];
        switch (currentChar)
        {
        case '(':
        case '[':
        case '{':
            validatorStack->push(currentChar);
            break;

        case ')':
        case ']':
        case '}':
            if (validatorStack->isEmpty()) {
                return false;
            } else {
                char charInStack = validatorStack->pop();
                if (currentChar == ')' && charInStack != '(')
                    return false;
                else if (currentChar == ']' && charInStack != '[')
                    return false;
                else if (currentChar == '}' && charInStack != '{')
                    return false;
            }
            break;
        
        default:
            break;
        }
    }
    return validatorStack->isEmpty();
}


int main() {
    std::cout << std::boolalpha;

    string file1 = "texto1.txt";
    string contentFile1 = readFileContent(file1);
    bool balancerFile1 = checkParenthesis(contentFile1);
    cout << "Archivo " << file1 << " - Balanceo: " << balancerFile1 << "\n";

    string file2 = "texto2.txt";
    string contentFile2 = readFileContent(file2);
    bool balancerFile2 = checkParenthesis(contentFile2);
    cout << "Archivo " + file2 + " - Balanceo: " << balancerFile2;
}

